### PR TITLE
HWY-{167, 205}: Create LNC-compliant units.

### PR DESCRIPTION
**Pull request by afck**
As discussed, this adds my suggestions on top of #572.

https://casperlabs.atlassian.net/browse/HWY-167
https://casperlabs.atlassian.net/browse/HWY-205

### CODE REVIEW

**Comment by goral09** [comment]
Since I worked on the original PR and didn't catch the errors you did, I am not sure I should be approving the PR. To me it obviously looks OK.

**Review by goral09 — APPROVED** [review]

**Comment by afck** [comment]
Also, `liveness_test_some_equivocate` just failed for me, so this isn't ready anyway. :pensive:

**Comment by goral09** [comment]
> I agree; I don't think any of the authors should approve. :-P

That greatly narrows down the set of people that _can_ approve it :)

**Comment by afck** [comment]
Closing this for now, since there are still unresolved issues.
Will keep pushing to the same branch, and reopen once it's ready to merge.

**Comment by goral09** [comment]
Why not turn it into a draft?

**Comment by afck** [comment]
It looks like you can't turn it back into a draft anymore. I can open a new draft PR, though.

File: node/src/components/consensus/highway_core/state.rs [hunk 1]
```diff
@@ -146,11 +146,8 @@ pub(crate) struct State<C: Context> {
     faults: HashMap<ValidatorIndex, Fault<C>>,
     /// The full panorama, corresponding to the complete protocol state.
     panorama: Panorama<C>,
-    /// Panorama used when creating new units.
-    /// In the presence of faults may lag behind `panorama`.
-    /// Units, that if cited by a new unit would make that unit violate the LNC, are not added to
-    /// `citable_panorama`.
-    citable_panorama: Panorama<C>,
+    /// The merged panoramas of all endorsed units. This is always safe to cite.
+    endorsed_panorama: Panorama<C>,
```
Comment (goral09) [inline node/src/components/consensus/highway_core/state.rs:150]: Are all units really endorsed or some just don't need to b/c they don't cite any equivocator?

Comment (goral09) [reply node/src/components/consensus/highway_core/state.rs:150]: I see, this has changed.

Comment (afck) [reply node/src/components/consensus/highway_core/state.rs:150]: Exactly: Those are really only the endorsed units and those cited by them.

**Comment by afck** [comment]
Closing this after all. I'm about to reshuffle some of the changes and undo others. Should be a smaller PR in the end.

**Pull request closed without merging**